### PR TITLE
Add start command to woocommerce/client/legacy

### PR DESCRIPTION
**Pull request by nathanss**
### All Submissions:

-   [ ] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Create a watch command for the legacy code. Previously we had to run build manually every time.

### How to test the changes in this Pull Request:

- [ ] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. Run the start command.
2. Open the legacy product manager.
3. Make a change to the code inside `plugins/woocommerce/client/legacy/js` directory.
4. Check if the build command runs automatically
5. Refresh the page and check if your change is there

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?
-   [ ] Have you included testing instructions?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/client/legacy/package.json [hunk 1]
```diff
@@ -7,7 +7,8 @@
 	"main": "Gruntfile.js",
 	"scripts": {
 		"turbo:build": "grunt assets",
-		"build": "pnpm -w exec turbo run turbo:build --filter=$npm_package_name"
+		"build": "pnpm -w exec turbo run turbo:build --filter=$npm_package_name",
+		"watch": "nodemon --watch js --watch css -e js,scss --exec pnpm build"
```
Comment (ObliviousHarmony) [inline plugins/woocommerce/client/legacy/package.json]: As far as I can tell, all of the other watch commands in our repository are `"start"`. We should be consistent with that here.

**Review by ObliviousHarmony — CHANGES_REQUESTED** [review]
I left a comment on the name of the script but otherwise I think this is a reasonable change. It tests well and does what it says on the tin.

As an aside, I've got [an open issue](https://github.com/woocommerce/woocommerce/issues/37377) that I plan to work on around creating a standardized `"watch"` command. Seeing the use of `nodemon` here is very interesting. I was under the impression that the tool could only run a Node server, but, apparently you can run arbitrary commands! Given that you can also watch file extensions, this might be a good alternative route to take on that issue.

**Comment by nathanss** [comment]
@ObliviousHarmony thanks for the review! I'm glad I have influenced you in a good way. 

I addressed your comments about renaming to 'start'!

**Review by ObliviousHarmony — APPROVED** [review]
Thanks again!

**Pull request merged**